base_layer/transaction_components/src/lib.rs

```diff
@@ -64,3 +67,12 @@ impl BanReason {
         self.ban_duration
     }
 }
+
+// Helper function to derive a TxId from the first ransaction output, or a random TxId if there are no outputs
+pub(crate) fn tx_outputs_to_tx_id(mac_key: &[u8], outputs: &[TransactionOutput]) -> TxId {
+    if let Some(first_output) = outputs.first() {
+        TxId::new_deterministic(mac_key, &first_output.hash())
+    } else {
+        TxId::new_random()
```

Comment:
You should create each output's hash, sort and hash them together.

Reply:
The order when scanning or the order here, is not the same, but also there is no order guarantee when scanning

Reply:
Agreed,
 _the tx_id assigned to the first recovered output of that transaction would always correspond to the original tx_id_ 
 should read 
 _when doing recovery and hitting the first ordered output of the original transaction, the tx_id assigned to that recovered output would always correspond to the original tx_id_